### PR TITLE
Feature/121789 send message from job time series eventhub

### PR DESCRIPTION
**Pull request by ftoft**
## Description

Initial implementation of integration for gridloss and system correction timeseries, back to timeseries servicebus.
This is best effort, because timeseries service bus is not ready yet.

## References

* #0000

### CODE REVIEW

File: source/coordinator/GreenEnergyHub.Aggregation.Domain/GreenEnergyHub.Aggregation.Domain.csproj [hunk 1]
```diff
@@ -29,6 +29,31 @@ limitations under the License.
     </Content>
   </ItemGroup>
 
+  <ItemGroup>
+    <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed">
+      <HintPath>..\..\..\..\..\..\..\AppData\Roaming\JetBrains\Rider2020.3\azure-functions-coretools\3.0.3442\Newtonsoft.Json.dll</HintPath>
```
Comment (kristianschneider) [inline source/coordinator/GreenEnergyHub.Aggregation.Domain/GreenEnergyHub.Aggregation.Domain.csproj:34]: What is using newtonsoft?

**Review by kristianschneider — APPROVED** [review]
Looks OK. Note my question

**Review by kristianschneider — CHANGES_REQUESTED** [review]
I have a single question. Otherwise I think it is good to proceed

File: source/coordinator/GreenEnergyHub.Aggregation.Domain/GreenEnergyHub.Aggregation.Domain.csproj (hunk 1 above)
Comment (ftoft) [reply source/coordinator/GreenEnergyHub.Aggregation.Domain/GreenEnergyHub.Aggregation.Domain.csproj]: I was not used. I have removed it.

**Review by kristianschneider — APPROVED** [review]

File: source/coordinator/GreenEnergyHub.Aggregation.Domain/MeteringPointMessage.proto [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ */
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+option csharp_namespace = "GreenEnergyHub.Aggregation.Domain";
+
+message MeteringPointMessage {
+  string mRID = 1;
```
Comment (jonasdmoeller) [inline source/coordinator/GreenEnergyHub.Aggregation.Domain/MeteringPointMessage.proto:23]: Properties should start with capital letter.

File: source/coordinator/GreenEnergyHub.Aggregation.Infrastructure/ServiceBusProtobuf/ServiceBusChannelBase.cs [hunk 3]
```diff
@@ -24,14 +24,14 @@
 
 namespace GreenEnergyHub.Aggregation.Infrastructure.ServiceBusProtobuf
 {
-    public class ServiceBusChannel : Channel, IAsyncDisposable
+    public abstract class ServiceBusChannelBase<T> : Channel, IAsyncDisposable
```
Comment (jonasdmoeller) [inline source/coordinator/GreenEnergyHub.Aggregation.Infrastructure/ServiceBusProtobuf/ServiceBusChannelBase.cs:27]: Did you see the other two implementation of the service bus channel (from batman and volt)? We think we could simplify this a lot.

**Review by jonasdmoeller — CHANGES_REQUESTED** [review]
Have you considered writing unit testing the introduced logic?

File: source/coordinator/GreenEnergyHub.Aggregation.Coordinator/Startup.cs [hunk 4]
```diff
@@ -78,9 +78,12 @@ public override void Configure(IFunctionsHostBuilder builder)
             };
 
             builder.Services.AddSingleton(coordinatorSettings);
-            builder.Services.AddSingleton<Channel>(x => new ServiceBusChannel(connectionStringServiceBus, "aggregations", x.GetRequiredService<ILogger<ServiceBusChannel>>()));
-
+            builder.Services.AddSingleton(x => new PostOfficeServiceBusChannel(connectionStringServiceBus, "aggregations", x.GetRequiredService<ILogger<PostOfficeServiceBusChannel>>()));
+            //TODO: This configuration of servicebus is only for testing and need to be corrected, when timeseries domain is ready
+            builder.Services.AddSingleton(x => new TimeseriesServiceBusChannel(connectionStringServiceBus, "timeseries", x.GetRequiredService<ILogger<TimeseriesServiceBusChannel>>()));
```
Comment (ZavezX) [inline source/coordinator/GreenEnergyHub.Aggregation.Coordinator/Startup.cs:83]: Naming is off here as well, should be `TimeSeriesServiceBusChannel`.

**Review by ZavezX — CHANGES_REQUESTED** [review]

File: source/coordinator/GreenEnergyHub.Aggregation.Domain/MeteringPointMessage.proto (hunk 2 above)
Comment (johevemi) [reply source/coordinator/GreenEnergyHub.Aggregation.Domain/MeteringPointMessage.proto:23]: this is a proto file and it should reflect the real name of the property. The autogenerated code from this file will change it automatically from (mRID) to match .NET standards (MRID)

**Review by ZavezX — APPROVED** [review]

File: source/coordinator/GreenEnergyHub.Aggregation.Infrastructure/ServiceBusProtobuf/ServiceBusChannelBase.cs (hunk 3 above)
Comment (ftoft) [reply source/coordinator/GreenEnergyHub.Aggregation.Infrastructure/ServiceBusProtobuf/ServiceBusChannelBase.cs:27]: No. We have not looked in to it.

File: source/coordinator/GreenEnergyHub.Aggregation.Application/Coordinator/Strategies/CombinedGridLossStrategy.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            TimeseriesDispatcher timeseriesDispatcher)
+        : base(logger, timeseriesDispatcher)
+        {
+            _glnService = glnService;
+            _specialMeteringPointsService = specialMeteringPointsService;
+        }
+
+        public string FriendlyNameInstance => "combined_grid_loss";
+
+        public override IEnumerable<IOutboundMessage> PrepareMessages(
```
[the remark this answers is not in the document]
Comment (ftoft) [reply source/coordinator/GreenEnergyHub.Aggregation.Application/Coordinator/Strategies/CombinedGridLossStrategy.cs:51]: Maybe it is a good idea, but can we take it in a other task?

**Pull request merged**